app/views/spree/admin/orders/invoice.html.haml

```diff
@@ -26,30 +26,42 @@
         = @order.distributor.address.full_address
         %br
         = @order.distributor.contact.email
-      %td{width: "10%" }
+    %tr{ valign: "top" }
+      %td{ colspan: 3 }
         &nbsp;
-      %td{ :align => "right" }
-        %strong= "#{t('.to')}: #{@order.ship_address.full_name}"
+    %tr{ valign: "top" }
+      %td{ align: "left" }
+        %strong= "#{t('.to')}:"
+        %br
+        = @order.bill_address.full_name
         - if @order.andand.customer.andand.code.present?
           %br
           = "#{t('.code')}: #{@order.customer.code}"
         %br
-        = @order.ship_address.full_address
+        = @order.bill_address.full_address
         %br
         - if @order.andand.customer.andand.email.present?
           = "#{@order.customer.email},"
         = "#{@order.bill_address.phone}"
+      %td
+        &nbsp;
+      %td{ align: "left", style: "border-left: .1em solid black; padding-left: 1em" }
+        %strong= "#{t('.shipping')}: #{@order.shipping_method&.name}"
+        - if @order.shipping_method&.require_ship_address
+          %br
+          = @order.ship_address.full_name
+          %br
+          = @order.ship_address.full_address
+          %br
+          = @order.ship_address.phone
+        - if @order.special_instructions.present?
+          %br
+          %br
+            %strong= t :customer_instructions
```

Comment:
we have parenthesis everywhere in the code in cases like this, I'd keep with local idiom
t(:customer_instructions)

EDIT: I now see you just copied it from below :-)